h2o-core/src/main/java/water/parser/Parser.java

```diff
@@ -87,7 +87,10 @@ ParseWriter streamParseZip( final InputStream is, final StreamParseWriter dout,
     StreamParseWriter nextChunk = dout;
     int zidx = bvs.read(null,0,0); // Back-channel read of chunk index
     assert zidx==1;
-    while( is.available() > 0 ) {
+    int streamAvailable = is.available();
+    while(streamAvailable > 0) {
+      parseChunk(cidx++, din, nextChunk);
+      streamAvailable = is.available(); // Can (also!) rollover to the next input chunk
```

Comment:
I do not understand the comment - how can `available` call fetch another chunk?

Do you mean, that parseChunk can cross chunk boundaries, and as a side effect `is.available` is changed?